sphinx_gallery/gen_rst.py

```diff
@@ -410,17 +400,23 @@ def scale_image(in_fname, out_fname, max_width, max_height):
                           generated images')
 
 
-def save_thumbnail(thumbnail_image_path, base_image_name, gallery_conf):
+def save_thumbnail(thumbnail_image_path, src_file, gallery_conf):
     """Save the thumbnail image"""
     thumb_dir = os.path.join(os.path.dirname(thumbnail_image_path), 'thumb')
     if not os.path.exists(thumb_dir):
         os.makedirs(thumb_dir)
 
+    base_image_name = os.path.splitext(os.path.basename(src_file))[0]
     thumb_file = os.path.join(thumb_dir,
                               'sphx_glr_%s_thumb.png' % base_image_name)
 
-    if os.path.exists(thumbnail_image_path):
+    if src_file in gallery_conf['failing_examples']:
```

Comment:
Just to be clear, what I meant by "doesn't seem to work", the broken image is not shown in the gallery:
![sphinx-gallery](https://cloud.githubusercontent.com/assets/1680079/15740513/6a8caf5c-28b5-11e6-8066-a4f4c866c1e8.png)